src/process_wrap.cc

```diff
@@ -251,23 +249,26 @@ class ProcessWrap : public HandleWrap {
     if (!js_options->Get(context, env->args_string()).ToLocal(&argv_v)) {
       return;
     }
-    if (!argv_v.IsEmpty() && argv_v->IsArray()) {
+    std::vector<char*> options_args;
+    std::vector<std::string> args_vals;
+    if (argv_v->IsArray()) {
       Local<Array> js_argv = argv_v.As<Array>();
       int argc = js_argv->Length();
       CHECK_LT(argc, INT_MAX);  // Check for overflow.
-
-      // Heap allocate to detect errors. +1 is for nullptr.
-      options.args = new char*[argc + 1];
+      args_vals.reserve(argc);
+      options_args.resize(argc + 1);
       for (int i = 0; i < argc; i++) {
         Local<Value> val;
         if (!js_argv->Get(context, i).ToLocal(&val)) {
           return;
         }
         node::Utf8Value arg(env->isolate(), val);
-        options.args[i] = strdup(*arg);
-        CHECK_NOT_NULL(options.args[i]);
+        args_vals.emplace_back(arg.ToString());
+        options_args[i] = const_cast<char*>(args_vals.back().c_str());
+        CHECK_NOT_NULL(options_args[i]);
```

Comment:
This assumes that the pointer returned by `args_vals.back().c_str()` will not be invalidated by a subsequent `args_vals.emplace_back(arg.ToString())` call. I think this is _technically_ implied by the `.reserve()` call, but nevertheless feels very brittle. So I'd suggest filling `options_args` only _after_ `args_vals` is no longer mutated, and not while we are emplacing elements.

(Also, I think you could make this a `std::vector<node::Utf8Value>` too, which would be one less allocation anyway?)

Reply:
Thanks for the review! I’ve applied your suggestions.

(Regarding `node::Utf8Value`: it isn’t copyable/movable, so it seems it can’t be stored in a `std::vector` (see [https://github.com/nodejs/node/pull/56811](https://github.com/nodejs/node/pull/56811)). Because of that, the extra allocation is unavoidable in this case.)